crates/iota-sdk-types/src/transaction/mod.rs

```diff
@@ -1118,9 +1157,47 @@ pub enum Argument {
 }
 
 impl Argument {
-    /// Turn a Result into a NestedResult. If the argument is not a Result,
-    /// returns None.
-    pub fn nested(&self, ix: u16) -> Option<Argument> {
+    crate::def_is!(Gas, Input, Result, NestedResult);
+
+    pub fn as_input_opt(&self) -> Option<u16> {
```

Comment:
Why not declare these through `def_is_as_into_opt`?

Reply:
So the reason is they are `Copy` types and the macro would return a `&u16`

Reply:
Hmm ok I see